scripts/lib/CIME/XML/component.py

```diff
@@ -116,14 +118,74 @@ def _get_value_match(self, node, attributes=None, exact_match=False):
         return match_value
 
     #pylint: disable=arguments-differ
-    def get_description(self, compsetname):
+    def get_description(self, compsetname, comp_class=None):
```

Comment:
Again, please add description of the comp_class argument.

Also, since comp_class is passed into the `__init__` method, is it possible to store comp_class as self._comp_class so you don't need to pass it in here, too? At a glance, it looks like it only needs to be passed in for drv to distinguish between the forcing and something else; otherwise you typically pass the same comp_class here as was used in `__init__`; if so that should be documented. Or maybe even more clear would be to introduce a separate get_description_forcing method, since it looks like the code path differs pretty significantly when comp_class is 'forcing' vs. anything else.